components/Chips/src/MDCChipView.h

```diff
@@ -0,0 +1,169 @@
+/*
+ Copyright 2017-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <UIKit/UIKit.h>
+
+@interface MDCChipView : UIControl
+
+/*
+ A UIImageView that leads the title label.
+
+ Only visible when the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) UIImageView *imageView;
+
+/*
+ A UIImageView that leads the title label. Appears above the imageView.
+
+ Only visible when the chip is selected and the image is non-nil.
+ */
+@property(nonatomic, readonly, nonnull) UIImageView *selectedImageView;
+
+/*
+ A UIView that trails the title label.
+
+ It will be given a size based on the value returned from sizeThatFits:.
+ */
+@property(nonatomic, strong, nullable) UIView *accessoryView;
+
+/*
+ The title label.
+
+ @note The title color is controlled by setTitleColor:forState:.
+ */
+@property(nonatomic, readonly, nonnull) UILabel *titleLabel;
+
+/*
+ Padding around the image view. Only used if the image view has a non-nil image.
+
+ Defaults to (0, 0, 0, 0).
+ */
+@property(nonatomic, assign) UIEdgeInsets imagePadding UI_APPEARANCE_SELECTOR;
```

Comment:
"Padding" isn't a concept that UIKit APIs use as far as I can tell - would it change the meaning to hew closer to UIKit's established conventions and call these "imageInsets", "titleInsets" and "accessoryInsets"?

Reply:
I thought about this a bit and at the time decided "padding" made the most sense. In UIKit insets aren't always synonymous with padding so it isn't ideal IMO. For example UIButton titleEdgeInsets aren't used in sizing (but contentEdgeInsets are). That said, I'm open to discussion.